### PR TITLE
Feature/add discount

### PR DESCRIPTION
**Pull request by UlanKubanychbekov**

### CODE REVIEW

File: src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
+public class Discount {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private int percent;
-    private LocalDate dateOfStart;
-    private LocalDate dateOfFinish;
-    @OneToOne(mappedBy = "discount")
-    private Product product;
+    private LocalDate startOfDate;
```
Comment (peaksoft-school) [inline src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java:23]: dateOfStart

File: src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java [hunk 2]
```diff
@@ -1,32 +1,38 @@
...
+    private LocalDate finishOfDate;
```
Comment (peaksoft-school) [inline src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java:24]: dateOfFinish

File: src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
 @Getter
 @Setter
 @NoArgsConstructor
-public class Discount {
 
+public class Discount {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private int percent;
```
Comment (peaksoft-school) [inline src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java:22]: percent int bolobu?

Comment (UlanKubanychbekov) [reply src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java:22]: да, скидка интовый, продукты тоже инт

File: src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java [hunk 4]
```diff
@@ -1,32 +1,38 @@
...
 import lombok.Setter;
-
 import javax.persistence.*;
 import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
 
 @Entity
-@Table(name = "discounts")
+@Table(name = "discount")
```
Comment (peaksoft-school) [inline src/main/java/kg/gadgetarium/gadgetarium/db/model/Discount.java:13]: why did you change it?

File: src/main/java/kg/gadgetarium/gadgetarium/apis/DiscountApi.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+@RequiredArgsConstructor
+@CrossOrigin
+@Tag(name = "discount", description = "admin create discount")
+public class DiscountApi {
+
+    private final DiscountService discount;
+
+    @PostMapping
+    @Operation(summary = "Discount Api", description = "admin can create discount")
+    public SimpleResponse create(RequestDiscount requestDiscount) {
```
Comment (peaksoft-school) [inline src/main/java/kg/gadgetarium/gadgetarium/apis/DiscountApi.java:25]: We should return discountResponse not **SimpleResponse**

**Review by peaksoft-school — COMMENTED** [review]
Commented 10/08/22 at 5.35 pm
